prov/verbs/src/verbs_info.c

```diff
@@ -1401,8 +1401,11 @@ static int fi_ibv_get_match_infos(uint32_t version, const char *node,
 
 	// TODO check for AF_IB addr
 	ret = fi_ibv_get_matching_info(version, hints, info, *raw_info,
+				       (hints && (hints->addr_format == FI_SOCKADDR ||
+						  hints->addr_format == FI_SOCKADDR_IN ||
+						  hints->addr_format == FI_SOCKADDR_IN6)) ?
```

Comment:
Since this patch originates from Dmitry, we need his signed-off-by line included.

I'd like to push the proper check into ofi_is_wildcard_listen_addr().  For the check itself, I think we should try to handle the no-hint case without assuming that the node/service parameters *won't* map to a valid sockaddr.  That is modify ofi_is_wildcard_listen_addr to:
```
if (hints && hints->addr_format != FI_SOCKADDR &&
    hints->addr_format != FI_SOCKADDR_IN &&
    hints->addr_format != FI_SOCKADDR_IN6))
        return 0;

/* else it's okay to call getaddrinfo, proceed with processing */
```